public/js/length_visualisation.js

```diff
@@ -0,0 +1,247 @@
+import _ from 'underscore';
+import * as Helpers from './visualisation_helpers';
+
+export default class LengthVisualisation {
+  constructor(query, $svg_container, algorithm) {
+    this._svg_container_d3 =  d3.select($svg_container[0]);
+    this.query = query;
+    this._seq_type = Helpers.get_seq_type(algorithm);
+
+    this._margin = {top: 30, right: 60, bottom: 70, left: 25};
+    this._width = $svg_container.width() - this._margin.left - this._margin.right;
+    this._height = 400 - this._margin.top - this._margin.bottom;
+
+    this._data = this.fill_data();
+    this._x_scale = d3.scale.linear()
+        .domain([
+          0,
+          (d3.max([this.query.length,d3.max(this._data)]) * 1.05)
+        ])
+        .range([0, this._width]);
+    this._bins = this.create_bins();
+    this._update_data = this.update_data();
+    this._y_scale = d3.scale.linear()
+        .domain([0, d3.max(this._bins, function(d) { return d.length })])
+        .range([this._height, 0]);
+    this._axis = this.create_axes();
+    this.svg = this._svg_container_d3.insert('svg', ':first-child')
+        .attr('width', this._width + this._margin.right + this._margin.left)
+        .attr('height', this._height + this._margin.top + this._margin.bottom)
+        .append('g')
+        .attr('transform','translate('+this._margin.left+','+this._margin.top+')');
+    this.setupTooltip();
+    this.draw_visualisation();
+  }
+
+  fill_data() {
+    // var data = [];
+    // this.query.hits.map(function (d) {
+    //   data.push(d.length)
+    // })
+    // return data;
+    return _.map(this.query.hits, _.iteratee('length'))
+  }
+
+  setupTooltip() {
+    $('[data-toggle="tooltip"]').tooltip({
+      'placement': 'top',
+      'container': 'body',
+      'html': 'true',
+      'delay': 0,
+      'white-space': 'nowrap'
+    });
+  }
+
+  tick_formatter(seq_type) {
+    var ticks = this._x_scale.ticks();
+    var format = d3.format('.1f');
+    var prefix = d3.formatPrefix(ticks[ticks.length - 1]);
+    var suffixes = {amino_acid: 'aa', nucleic_acid: 'bp'};
+    return function (d) {
+      if (d === 0) { return ; }
+      if (_.indexOf(ticks,d) >= 0) {
+        if (suffixes[seq_type] == 'aa') {
+          return (d + ' ' + suffixes[seq_type])
+        } else {
+          return (format(prefix.scale(d)) + ' ' + prefix.symbol + suffixes[seq_type]);
+        }
+      } else {
+        return ;
+      }
+    };
+  }
+
+  create_bins() {
+    var bins = d3.layout.histogram()
+        .range(this._x_scale.domain())
+        .bins(this._x_scale.ticks(50))
+        (this._data);
+    return bins;
+  }
+
+  prettify_evalue(evalue) {
```

Comment:
This functionality is needed in at least 4 places that I can remember: tabular summary, alignment overview (tooltip), stats table before pairwise alignment. And now length distribution. Could you extract this into a helper method and use the same throughout. Treat `report.js`' `prettify_float` to be authoritative.

Similarly, tooltip generation should be shared between this and alignment overview.